### DISCUSSION TITLE
Export geojson format for detection output

### DISCUSSION THREAD

**Discussion by lehen20** [azavea/raster-vision#2241] [Q&A]
I am using lightning model for training of tif file. 
1. The results i get are in tif format (scores.tif). Is there a way i can export them in geojson directly. 
2. In case i try to train on a big tif (approx 18gb), it takes a lot of time to train. what approach can i use for faster training.

**Comment by AdeelH** [comment]
>The results i get are in tif format (scores.tif). Is there a way i can export them in geojson directly.

You can have RV output geojson predictions by passing `vector_outputs` to [`SemanticSegmentationLabels.save()`](https://docs.rastervision.io/en/stable/usage/tutorials/pred_and_eval_ss.html#Save-predictions-to-file----SemanticSegmentationSmoothLabels.save()). Example:

```py
from rastervision.core.data import PolygonVectorOutputConfig

pred_labels.save(
    uri=f'out/',
    crs_transformer=crs_transformer,
    class_config=class_config,
    # VectorOutputConfigs for each class you want outputted as GeoJSON
    vector_outputs=[
        PolygonVectorOutputConfig(
            class_id=class_config.get_class_id('my_class'),
            # threshold to use for creating binary mask that will then be vectorized
            # (default: 1 / num_classes)
            threshold=0.5,
            # filter out positive regions in the binary mask smaller than
            # this diameter (in pixels)
            denoise=8,
        ),
    ],
)
```
See docs for `PolygonVectorOutputConfig` [here](https://docs.rastervision.io/en/stable/api_reference/_generated/rastervision.core.data.label_store.semantic_segmentation_label_store_config.PolygonVectorOutputConfig.html).

>In case i try to train on a big tif (approx 18gb), it takes a lot of time to train. what approach can i use for faster training.

- The more data you train on (i.e. the larger your scene), the longer it will take. If you are currently training with sliding windows but don't think training on all of them is necessary, you could [try sampling random windows](https://docs.rastervision.io/en/stable/usage/tutorials/sampling_training_data.html#RandomWindowGeoDataset) instead and use `max_windows` to limit the number of chips per epoch.
- If you are accessing the TIFF remotely, downloading it first might also speed up reading.
- Using a different or smaller model might also help. If you are currently using the model (DeepLab) in the [Lightning tutorial](https://docs.rastervision.io/en/stable/usage/tutorials/lightning_workflow.html), you could try the one in the [Training tutorial](https://docs.rastervision.io/en/stable/usage/tutorials/train.html#Define-model) instead.

**Comment by lehen20** [comment]
Thanks for your response!
